### PR TITLE
Fix external enums not assignable as constants

### LINKED ISSUE #70213: GDScript 2.0: Cannot assign external enum as a constant
### Godot version

4.0.dev (10bc1d8)

### System information

Linux (Ubuntu 22.04), Vulkan, Radeon RX Vega 64 (eGPU)

### Issue description

Given the following GDScript file:
```gdscript
extends Node

const External: = preload("res://external.gd")
const MyEnum: = External.MyEnum
```

Godot will return this error:
```
SCRIPT ERROR: Parse Error: Initializer for a constant must be a constant expression.
          at: GDScript::reload **(res://main.gd:4)**
```

It should be possible to assign external enums as constants.

Seems to be a regression caused by https://github.com/godotengine/godot/pull/69907. The issue doesn't exhibit in beta 8.

Similar to #69917, which is about external inner classes.

### Steps to reproduce

1. Open minimal project
2. Open `main.gd`
3. See that Godot returns the previously mentioned error

### Minimal reproduction project

[external_enum_constant.zip](https://github.com/godotengine/godot/files/10252654/external_enum_constant.zip)

### PR DESCRIPTION
**Pull request by adamscott**
Make external enums assignable as constants.

Also:
- Add external enums test
- Rename external inner class test
- Clean up `GDScriptAnalyzer::reduce_identifier_from_base` class behavior

Fixes #70213 - GDScript 2.0: Cannot assign external enum as a constant

### CODE REVIEW

File: modules/gdscript/tests/scripts/analyzer/features/external_enum_as_constant.gd [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const External = preload("external_enum_as_constant_external.notest.gd")
+const MyEnum = External.MyEnum
+
+func test():
+    print(MyEnum.WAITING == 0)
+    print(MyEnum.GODOT == 1)
```
Comment (akien-mga) [inline modules/gdscript/tests/scripts/analyzer/features/external_enum_as_constant.gd:6]: Missing newline at EOF, so it fails style checks. (Same for the `notest` one.)

**Comment by akien-mga** [comment]
CC @rune-scape

File: modules/gdscript/tests/scripts/analyzer/features/external_enum_as_constant.gd (hunk 1 above)
Comment (adamscott) [reply modules/gdscript/tests/scripts/analyzer/features/external_enum_as_constant.gd:6]: Thanks!

**Comment by anvilfolk** [comment]
Generally looks good to me!

**Review by anvilfolk — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**